selendroid-server/src/main/java/io/selendroid/server/action/ActionChain.java

```diff
@@ -18,10 +18,11 @@
   private int pauseTime = 0;
   private int index = 0;
 
-  public ActionChain(JSONObject actionChain) throws JSONException {
+  public ActionChain(JSONObject actionChain, int i) throws JSONException {
     this.inputDevice = actionChain.getString("inputDevice");
     this.actionChain = actionChain.getJSONArray("actions");
     this.context = new ActionContext();
+    this.context.setId(i);
```

Comment:
We could use [UUID's hashcode](http://www.tutorialspoint.com/java/util/uuid_hashcode.htm).  For the purposes here, the 32 bits of randomness should be enough.  The one in many million chance that a collision does happen it might make for a confusing test run.

Incrementing a class variable initialized to 0 perhaps could work as @chooper9 mentioned also.